[quant] Add support for Quantized Conv1d and ConvRELU1d

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38449 [quant] Support for functional quantized::conv1d
* #38441 [quant][graphmode] Add support for quantized conv1d + relu fusion
* #38438 [quant] Fusion support for conv1d + ReLU
* #38407 [quant] Return default qconfig when backend is 'none'
* #38341 [quant][graphmode] Add quantized::conv1d to graphmode
* #38332 [quant] Add quantized::conv1d op benchmarck
* **#38283 [quant] Add support for Quantized Conv1d and ConvRELU1d**
* #38248 [quant] Add support for quantized::conv1d operator
* #38247 [quant] Implement unsqueeze/squeeze for per-channel qtensor

Summary:
Adds support for the modules and tests

Test Plan:
python test/test_quantization.py TestStaticQuantizedModule.test_conv1d_api

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21553665](https://our.internmc.facebook.com/intern/diff/D21553665)

supriyar committed:
[quant] Add support for Quantized Conv1d and ConvRELU1d
Summary:
Adds support for the modules and tests

Test Plan:
python test/test_quantization.py TestStaticQuantizedModule.test_conv1d_api

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -186,8 +186,6 @@ class Conv1d(_ConvNd):
 
-    # We are using Conv2d to run the Conv1d. For that we need to know which
-    # dimension is squeezed/unsqueezed.
     _SQUEEZE_DIM = -2  # -2 is faster than -1.
```
what if you remove this

supriyar replied:
Yeah this is the culprit. I kept it around since I had to call `_packed_params.unpack()` (which calls conv2d unpack) so the output weight had to be squeezed for Conv1d. Let me directly try calling quantized::conv1d_unpack op instead.

supriyar committed:
Update on "[quant] Add support for Quantized Conv1d and ConvRELU1d"


Summary:
Adds support for the modules and tests

Test Plan:
python test/test_quantization.py TestStaticQuantizedModule.test_conv1d_api

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Add support for Quantized Conv1d and ConvRELU1d"


Summary:
Adds support for the modules and tests

Test Plan:
python test/test_quantization.py TestStaticQuantizedModule.test_conv1d_api

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on aten/src/ATen/native/quantized/cpu/qconv_unpack.cpp:
```diff
@@ -178,8 +178,9 @@ class QConv1dUnpackWeightsInt8 final {
       std::tie(weight, bias) = packed_weight->unpack();
-      weight = weight.squeeze_(quant_utils::kConv1dSqueezeDim + 2);
-      return std::tuple<at::Tensor, c10::optional<at::Tensor>>(weight, bias);
+      at::Tensor new_weight = weight.clone();
+      new_weight = new_weight.squeeze_(quant_utils::kConv1dSqueezeDim + 2);
+      return std::tuple<at::Tensor, c10::optional<at::Tensor>>(new_weight, bias);
```
does this change belong to pervious PR?

supriyar committed:
Update on "[quant] Add support for Quantized Conv1d and ConvRELU1d"


Summary:
Adds support for the modules and tests

Test Plan:
python test/test_quantization.py TestStaticQuantizedModule.test_conv1d_api

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
